### PR TITLE
multitenant: allow secondary tenants to split/scatter by default

### LINKED ISSUE #96736: capabilities: update the definition of the protobuf so that the zero protobuf struct results in pre-23.1 serverless behavior
**Describe the problem**

Once https://github.com/cockroachdb/cockroach/pull/96390 lands, the tenant capabilities subsystem will be hooked up. As a result, all secondary tenant requests will be subject to capability checks. As of the time of writing, the only `AdminRequest` that has capabilities associated with it is `AdminSplit`. However, this will soon expand to other operations such as scatters, relocate range, relocate lease etc. 

Internally, operations such as RESTORE/IMPORT etc. use these Admin requests. We will need to do something to prevent these operations from failing for existing (Serverless) tenants after upgrading to 23.1. 

(An option that was discussed internally, is to add a migration that grants secondary tenants `can_admin_split` (and any other capability they need that is added after writing up this issue) in `Serverless`. )

We are considering instead flipping the boolean meaning of the capabilities for pre-23.1 operations so that their default value coincides with what CC Serverless needs.

Jira issue: CRDB-24300
Epic:  CRDB-23344

### PR DESCRIPTION
**Pull request by arulajmani**
AdminSplit and AdminScatter requests are subject to capability checks.
Previously, these capabilities were codified in the "enabled" form. As
such, by default, secondary tenants did not have the ability to perform
these operations. This is in violation of what secondary tenants could
do prior to 23.1, at a time before capabilities existed. Moreover,
RESTORE/IMPORT rely on performing these operations for performance.
This made disallowing these operations by default a performance
regression.

This patch flips the phrasing of how these capabilities are stored on
the proto to use the "disable" verbiage. As such, secondary tenants are
able to perform splits and scatters by default. However, no change is
made to the public interface -- users above the `tenantcapabilitiespb`
package continue to interact with these capabilities as they were
before, oblivious to how these things are stored on disk.

There's a few testing changes here:
- As part of this change, we also clean up a testing knob that was used
by various backup, CDC, and logictests to override capability checks in
the authorizer. This isn't required with the new default behaviour.
- We also add some missing E2E tests for the `CanAdminUnsplit` capability
which were missing when it was introduced.

Fixes https://github.com/cockroachdb/cockroach/issues/96736

Release note: None

### CODE REVIEW

**Review by knz — COMMENTED** [review]
I like the simplification here.

However, I'd like to request one more step. It's good that we inverted the meaning in the protobuf, but IMHO the public interface should remain unchanged. See my comments below.

> Reviewed 36 of 36 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-:-NQSPdkx9ZwqSUGLwUrn:b248d39)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @benbardin, @ecwall, and @rharding6373)

___
*[`pkg/multitenant/tenantcapabilities/capabilities.go` line 165 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSPnlV6_DWvWcUIob6:-NQSPnlV6_DWvWcUIob7:b3b1clu) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/capabilities.go#L165)):*
> ```Go
> 	_ CapabilityID = iota
> 
> 	// DisableAdminScatter disallows a secondary tenant from being able to scatter
> ```

I recommend you undo the changes to this file. The original code is easier to read and understand.
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go` line 85 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSPtyH5pAzmH1jatb8:-NQSPtyH5pAzmH1jatb9:boafc1t) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go#L85)):*
> ```Go
> 			continue
> 		}
> 		switch request.Method() {
> ```

You can revert this change too. Let's keep all the checks under one conditional.
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go` line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSPzGyB0dUi1SOLAkl:-NQSPzGyB0dUi1SOLAkm:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go#L152)):*
> ```Go
> 	// The following are authorized via specific capabilities.
> 	kvpb.AdminScatter: tenantcapabilities.DisableAdminScatter,
> 	kvpb.AdminSplit:   tenantcapabilities.DisableAdminSplit,
> ```

ditto
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go` line 56 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSQ-Mx0ImMsIsnEd3B:-NQSQ-Mx0ImMsIsnEd3C:b8eli56) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go#L56)):*
> ```Go
> 	switch capabilityID {
> 	case tenantcapabilities.DisableAdminScatter:
> 		return boolCap{&t.DisableAdminScatter}
> ```

```go
case tenantcapabilities.CanAdminScatter:
   return invertedBoolCap{&t.DisableAdminScatter}
```
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go` line 58 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSQ2Vl7viKO70oN4Fx:-NQSQ2Vl7viKO70oN4Fy:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go#L58)):*
> ```Go
> 		return boolCap{&t.DisableAdminScatter}
> 	case tenantcapabilities.DisableAdminSplit:
> 		return boolCap{&t.DisableAdminSplit}
> ```

ditto
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go` line 76 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSQ3C165osgQa7rqkL:-NQSQ3C165osgQa7rqkM:b3pjiq7) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go#L76)):*
> ```Go
> 	switch capabilityID {
> 	case tenantcapabilities.DisableAdminScatter:
> 		return t.DisableAdminScatter
> ```

```go
case tenantcapabilities.CanAdminScatter:
   return !t.DisableAdminScatter
```

ditto below
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.proto` line 34 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSOaMB-R36WWkTeOIt:-NQSOaMB-R36WWkTeOIu:b-okhd2p) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.proto#L34)):*
> ```protobuf
> 
>   // DisableAdminSplit, if set to true, revokes the tenants ability to
>   // successfully perform `AdminSplit` requests.
> ```

I recommend you group the two disable fields close to each other in the proto definition then add a comment that explains why they are different from the rest. We must teach people coming after us that any new cap should be added at the end with a default-not-granted definition.

**Review by arulajmani — COMMENTED** [review]
Thanks for the suggestion. I quite like how this ended up looking after going with the `invertedBoolField` thing. RFAL. 

> Dismissed @knz from 5 discussions.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-:-NQSzIqQ81VL7RbXqqkB:b-nljrnp)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @benbardin, @ecwall, @knz, and @rharding6373)

___
*[`pkg/multitenant/tenantcapabilities/capabilities.go` line 165 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSPnlV6_DWvWcUIob6:-NQSqxou6WkHfPh8uH6J:byi0320) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/capabilities.go#L165)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

I recommend you undo the changes to this file. The original code is easier to read and understand.
</blockquote></details>

Done, but I kept some parts of the commentary. 
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go` line 85 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSPtyH5pAzmH1jatb8:-NQSpR8VCge6b3jeoZzG:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go#L85)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

You can revert this change too. Let's keep all the checks under one conditional.
</blockquote></details>

Done.
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go` line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSPzGyB0dUi1SOLAkl:-NQSpPd1-Ejm5VUl_TR7:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiesauthorizer/authorizer.go#L152)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go` line 56 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSQ-Mx0ImMsIsnEd3B:-NQSpNNq5jnprZWsY7Gr:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go#L56)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

```go
case tenantcapabilities.CanAdminScatter:
   return invertedBoolCap{&t.DisableAdminScatter}
```
</blockquote></details>

Done.
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go` line 58 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSQ2Vl7viKO70oN4Fx:-NQSpNvU9RpBRfXem98s:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go#L58)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go` line 76 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSQ3C165osgQa7rqkL:-NQSpLayDhMWY2p5OVe1:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.go#L76)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

```go
case tenantcapabilities.CanAdminScatter:
   return !t.DisableAdminScatter
```

ditto below
</blockquote></details>

Done.
___
*[`pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.proto` line 34 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQSOaMB-R36WWkTeOIt:-NQSpFbJ3Ffr5P7LZdUC:b-9hkgft) ([raw file](https://github.com/cockroachdb/cockroach/blob/57dd19c2a16e4f00be56108e17cd7fe787293ae0/pkg/multitenant/tenantcapabilities/tenantcapabilitiespb/capabilities.proto#L34)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

I recommend you group the two disable fields close to each other in the proto definition then add a comment that explains why they are different from the rest. We must teach people coming after us that any new cap should be added at the end with a default-not-granted definition.
</blockquote></details>

Done, and added commentary explaining the motivation behind the "disabled" verbiage.

**Review by ecwall — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @benbardin, @knz, and @rharding6373)

___
*[`pkg/sql/multitenant_admin_function_test.go` line 456 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVCimp6TLwGSm7l67U:-NQVCimp6TLwGSm7l67V:b-kv1czm) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L456)):*
> ```Go
> 				result: [][]string{{"\xf0\x89\x89", "/1", maxTimestamp}},
> 			},
> 			secondary: tenantExpected{
> ```

These `tenantExpected` blocks should not be removed since they are verifying output for certain tenant setups.
___
*[`pkg/sql/multitenant_admin_function_test.go` line 466 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVCqyzCaeNrWNwf7uL:-NQVCqz-5t89EJ8DNDun:bv6u4i7) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L466)):*
> ```Go
> 			},
> 			queryClusterSetting: sql.SecondaryTenantSplitAtEnabled,
> 			queryCapability:     tenantcapabilities.CanAdminSplit,
> ```

Change `queryCapability` to `type capability struct { id CapabilityID, value bool }` and set `value: false` for the split and scatter tests.
___
*[`pkg/sql/multitenant_admin_function_test.go` line 501 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVDjXp5uFDwp_S06Rs:-NQVDjXp5uFDwp_S06Rt:bba1sf8) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L501)):*
> ```Go
> 			},
> 			setupClusterSetting: sql.SecondaryTenantSplitAtEnabled,
> 			setupCapability:     tenantcapabilities.CanAdminSplit,
> ```

The `setupCapability` field can be removed from `type testCase struct` now because it was simulating split being enabled by default.

**Review by arulajmani — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @benbardin, @ecwall, @knz, and @rharding6373)

___
*[`pkg/sql/multitenant_admin_function_test.go` line 456 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVCimp6TLwGSm7l67U:-NQVL5HR1uazCq6u5e99:b6op75a) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L456)):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

These `tenantExpected` blocks should not be removed since they are verifying output for certain tenant setups.
</blockquote></details>

All these test cases that have been removed have functionally been moved to the datadriven tests in `pkg/ccl/multitenantccl/tenantcapabilitiesccl`. Now that we have that framework, it's a much better place to test stuff like this. 
___
*[`pkg/sql/multitenant_admin_function_test.go` line 466 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVCqyzCaeNrWNwf7uL:-NQVLJ9S5vFmrgwr0ZMZ:bb76mu4) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L466)):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

Change `queryCapability` to `type capability struct { id CapabilityID, value bool }` and set `value: false` for the split and scatter tests.
</blockquote></details>

Why?

**Review by ecwall — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @benbardin, @knz, and @rharding6373)

___
*[`pkg/sql/multitenant_admin_function_test.go` line 456 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVCimp6TLwGSm7l67U:-NQVN8mo11RVU7nYsWid:b3z6vbq) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L456)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

All these test cases that have been removed have functionally been moved to the datadriven tests in `pkg/ccl/multitenantccl/tenantcapabilitiesccl`. Now that we have that framework, it's a much better place to test stuff like this. 
</blockquote></details>

Can we discuss this change first? It looks like we are replacing data driven testing with a DSL that requires a lot of extra code.
___
*[`pkg/sql/multitenant_admin_function_test.go` line 466 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVCqyzCaeNrWNwf7uL:-NQVNUSr5NWujr65d0Aw:bmnpg7b) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L466)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Why?
</blockquote></details>

So that you can make the tests pass without removing test coverage here.

**Review by arulajmani — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @benbardin, @ecwall, @knz, and @rharding6373)

___
*[`pkg/sql/multitenant_admin_function_test.go` line 456 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVCimp6TLwGSm7l67U:-NQVOR8w0zNQwc1XuY1w:bcoq4rt) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L456)):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

Can we discuss this change first? It looks like we are replacing data driven testing with a DSL that requires a lot of extra code.
</blockquote></details>

I tried making changes to this test for a little bit before I gave up because of how cumbersome it was. As this code churns and stuff changes, the maintainability burden of this thing is high. I've personally felt it on 2 different occasions, so I finally decided to bite the bullet and move some of the things that were in my way to the datadriven tests. Those are much earier to read and easier to change/write as well.

I'm happy to get @knz to sign off on these changes if you're not happy with them. I do want to get this thing merged soon given it's a release blocker and overall this change is positive.  
___
*[`pkg/sql/multitenant_admin_function_test.go` line 466 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVCqyzCaeNrWNwf7uL:-NQVOGbqA-mSrchUCDRX:bfn8g9u) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L466)):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

So that you can make the tests pass without removing test coverage here.
</blockquote></details>

I see. Given we haven't lost any test coverage because things have merely moved, I think this is fine.

**Review by knz — APPROVED** [review]
As discussed elsewhere - either don't change the existing test framework structure (and _follow established patterns_) OR ensure this is done in a separate commit for clarity.

> Reviewed 18 of 18 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-:-NQVWDdL8TVagxW0ymeS:buhcnxj)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @benbardin, @ecwall, and @rharding6373)

___
*[`-- commits` line 27 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVRTIODxCYePe_uglf):*
nit cumbersome

**Review by arulajmani — COMMENTED** [review]
I pulled out the test movement changes into a separate commit, like we spoke about in other places. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-:-NQVdXxh-8frZTbau32D:bjd7apb)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @benbardin, @ecwall, @knz, and @rharding6373)

___
*[`pkg/sql/multitenant_admin_function_test.go` line 501 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVDjXp5uFDwp_S06Rs:-NQVdRNf7KGV3xf7cllV:b3ghg7f) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L501)):*
<details><summary><i>Previously, ecwall (Evan Wall) wrote…</i></summary><blockquote>

The `setupCapability` field can be removed from `type testCase struct` now because it was simulating split being enabled by default.
</blockquote></details>

We still make use of it, so it can't be removed.

**Review by arulajmani — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @benbardin, @ecwall, @knz, and @rharding6373)

___
*[`pkg/sql/multitenant_admin_function_test.go` line 501 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQVDjXp5uFDwp_S06Rs:-NQVp4Ze4dBVpK01KUXK:bp08rmn) ([raw file](https://github.com/cockroachdb/cockroach/blob/c31c1acce5832005b9ff7f65a3bc883b61c5ff6d/pkg/sql/multitenant_admin_function_test.go#L501)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

We still make use of it, so it can't be removed. 
</blockquote></details>

My comment no longer applies after the changes to the Authorizer. To add more words there, earlier, we couldn't remove it because we still needed for capabilities to propagate before making these changes -- `setupCapability` was providing us that mechanism. 

With the `Authorizer` changes, I tacked on a 3rd commit to clean this stuff up. Thanks for the pointer.

**Review by knz — COMMENTED** [review]
I generally like the 2nd commit. It's very good.

I am going to spend a minute more looking at the 1st and last commit. While I do this, please also contribute your thinking (with a concrete example) to #98483.

> Reviewed 33 of 33 files at r3, 34 of 35 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-:-NQVy7t07jepfXPJZmaf:b-6d7qe6)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @benbardin, @ecwall, and @rharding6373)

**Comment by knz** [comment]
I would like us to move this PR forward with the following change merged into it https://github.com/arulajmani/cockroach/pull/5  (will need a squash)

**Review by rharding6373 — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-:-NQWAIVP02fiAqtKjCfb:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @arulajmani, @benbardin, @ecwall, and @knz)

___
*[`pkg/ccl/multitenantccl/tenantcapabilitiesccl/testdata/can_admin_unsplit` line 24 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/98546#-NQW4Kbs8y0evDLmvkjr:-NQW4Kbs8y0evDLmvkjs:b-rxrm6b) ([raw file](https://github.com/cockroachdb/cockroach/blob/6c148e3ae7eb7e98f344c63208e74f6cc3e91b2c/pkg/ccl/multitenantccl/tenantcapabilitiesccl/testdata/can_admin_unsplit#L24)):*
> ```
> ALTER TABLE t UNSPLIT AT VALUES (0)
> ----
> pq: could not UNSPLIT AT (0): key /Tenant/10/Table/104/1/0 is not the start of a range
> ```

Could you also add a test that does a successful `UNSPLIT`?

**Review by ecwall — APPROVED** [review]

**Comment by knz** [comment]
We will now confirm this passes CI then we can merge the PR. Thanks!

**Comment by knz** [comment]
thanks to all for sharing the work on this!

bors r+
